client/src/components/FileUploadForm.vue

```diff
@@ -107,31 +107,43 @@ export default Vue.extend({
       return Object.keys(this.types);
     },
     createDisabled(): boolean {
-      return !this.file || !this.selectedType || !this.newTable || !!this.fileUploadError;
+      return (
+        !this.file ||
+        !this.selectedFileName ||
+        !this.selectedFileType ||
+        !!this.fileUploadError
+      );
     },
   },
 
   methods: {
     handleFileInput(file: File) {
       this.file = file;
-      const fileType = this.fileType(file);
 
-      if (fileType) {
-        this.selectedType = fileType;
+      if (!file) {
+        this.selectedFileName = null;
+        this.selectedFileType = null;
+        this.fileUploadError = null;
+        return;
+      }
+
+      const fileInfo = this.fileInfo(file);
+      if (fileInfo !== null) {
+        [this.selectedFileName, this.selectedFileType] = fileInfo;
```

Comment:
The autopopulation should only happen if the file name field is currently blank. The driving principle here is to never irreversibly lose any work the user may have done previously.

Reply:
Good point. I've just changed that.